### PR TITLE
devise_google_auth

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/???)

## WHAT (optional)

- Change number items `completed/total` in admin page.

## HOW

- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)

- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: app/models/tour.rb [hunk 1]
```diff
@@ -68,6 +69,14 @@ class Tour < ApplicationRecord
     where(avg_rating: get_rating) if get_rating.present?
   end)
 
+  scope :select_col, (lambda do |col|
+    select(col) if col.present?
+  end)
+
+  scope :join_tbl, (lambda do |tbl|
+    joins(tbl) if col.present?
+  end)
```
Comment (manhnt-0960) [inline app/models/tour.rb:78]: scope này dùng chỗ nào vậy e

File: app/models/user.rb [hunk 2]
```diff
@@ -20,6 +23,17 @@ class User < ApplicationRecord
 
   scope :lastest, ->{order(created_at: :desc)}
 
+  def self.from_omniauth access_token
```
Comment (manhnt-0960) [inline app/models/user.rb:26]: dùng class < self để khai báo class method

File: app/models/user.rb [hunk 3]
```diff
@@ -20,6 +23,17 @@ class User < ApplicationRecord
...
+    data = access_token.info
+    user = User.where(email: data["email"]).first
+    user || user = User.create(name: data["name"],
```
Comment (manhnt-0960) [inline app/models/user.rb:29]: sao k dùng method first_or_create e

**Pull request closed without merging**